src/app/app-error.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, ErrorHandler } from '@angular/core';
+import { ModalService } from './core';
+
+@Injectable()
+export class AppErrorService implements ErrorHandler {
+
+  private defaultHandler: ErrorHandler;
+
+  constructor(private modal: ModalService) {
+    this.defaultHandler = new ErrorHandler(false); // don't re-throw
+  }
+
+  handleError(err) {
+    this.modal.show({
+      title: 'Uncaught ' + (err.name || 'Error'),
+      body: `
+        <p>${err.message}</p>
+        <hr/>
+        <pre style="font-size:12px;line-height:16px;white-space:pre">${err.stack}</pre>
```

Comment:
Ew, I see what you mean, and now I like this pseudo template even less. :( That's not very helpful, but I will keep this in mind and think about potentially using a different approach.